packages/@react-spectrum/s2/src/Card.tsx

```diff
@@ -0,0 +1,669 @@
+  },
+  gridColumnEnd: {
+    ':nth-last-child(4):first-child': 'span 3'
+  },
+  objectFit: 'cover',
+  pointerEvents: 'none',
+  userSelect: 'none'
+});
+
+export const CollectionCardPreview = forwardRef(function CollectionCardPreview(props: CardPreviewProps, ref: DOMRef<HTMLDivElement>) {
+  let {size} = useContext(InternalCardContext)!;
+  return (
+    <CardPreview {...props} ref={ref}>
+      <div className={collection({size})}>
+        <ImageContext.Provider value={{styles: collectionImage}}>
+          {props.children}
+        </ImageContext.Provider>
+      </div>
+    </CardPreview>
+  );
+});
+
+export interface AssetCardProps extends Omit<CardProps, 'density'> {}
+
+export const AssetCard = forwardRef(function AssetCard(props: AssetCardProps, ref: DOMRef<HTMLDivElement>) {
+  return (
+    <Card {...props} ref={ref} density="regular">
+      <Provider
+        values={[
+          [ImageContext, {
+            alt: '',
+            styles: style({
+              width: 'full',
+              aspectRatio: 'square',
+              objectFit: 'contain',
+              pointerEvents: 'none',
+              userSelect: 'none'
+            })
+          }],
+          [IllustrationContext, {
+            render(icon) {
+              return (
+                <SkeletonWrapper>
+                  <div
+                    className={style({
+                      display: 'flex',
+                      alignItems: 'center',
+                      justifyContent: 'center',
+                      backgroundColor: 'gray-100',
+                      aspectRatio: 'square'
+                    })}>
+                    {icon}
+                  </div>
+                </SkeletonWrapper>
+              );
+            },
+            styles: style({
+              height: 'auto',
+              maxSize: 160,
+              // TODO: this is made up.
+              width: '[50%]'
+            })
+          }]
+        ]}>
+        {props.children}
+      </Provider>
+    </Card>
+  );
+});
+
+const avatarSize = {
+  XS: 24,
+  S: 48,
+  M: 64,
+  L: 64,
+  XL: 80
+} as const;
+
+export interface UserCardProps extends Omit<CardProps, 'density' | 'variant'> {
+  // Quiet is not supported due to lack of indent between preview and avatar.
+  variant?: 'primary' | 'secondary' | 'tertiary'
+}
+
+export const UserCard = forwardRef(function UserCard(props: CardProps, ref: DOMRef<HTMLDivElement>) {
+  let {size = 'M'} = props;
+  return (
+    <Card {...props} ref={ref} density="spacious">
+      <Provider
+        values={[
+          [ImageContext, {
+            alt: '',
+            styles: style({
+              width: 'full',
+              aspectRatio: '[3/1]',
+              objectFit: 'cover',
+              pointerEvents: 'none',
+              userSelect: 'none'
+            })
+          }],
+          [AvatarContext, {
+            size: avatarSize[size],
+            UNSAFE_style: {
+              '--size': avatarSize[size] + 'px'
```

Comment:
we've got to start an audit of all these variables we introduce

or hash them for internal module use or something. it'd be nice if we could export/import them internally as well, so we know where they are defined vs used

don't worry about in this PR